### PR TITLE
Remove unnecessary dependencies

### PR DESCRIPTION
**Pull request by marcospereira**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -18,26 +18,13 @@ val `play-java-grpc-example` = (project in file("."))
...
-javacOptions ++= List("-Xlint:unchecked", "-Xlint:deprecation", "-Werror")
-javacOptions -= "-Werror" // https://github.com/akka/akka-grpc/issues/490
+javacOptions ++= List("-Xlint:unchecked", "-Xlint:deprecation")
 
-libraryDependencies += guice
-
-// Test libraries
-val playVersion = play.core.PlayVersion.current
-val playGrpcVersion = "0.5.0"
-libraryDependencies += "com.lightbend.play"      %% "play-grpc-scalatest" % playGrpcVersion % Test
```
Comment (marcospereira) [inline build.sbt:29]: These are Scala dependencies. :-)

File: build.sbt [hunk 2]
```diff
@@ -18,26 +18,13 @@ val `play-java-grpc-example` = (project in file("."))
...
-libraryDependencies += "com.lightbend.play"      %% "play-grpc-specs2"    % playGrpcVersion % Test
-libraryDependencies += "com.typesafe.play"       %% "play-test"           % playVersion     % Test
-libraryDependencies += "com.typesafe.play"       %% "play-specs2"         % playVersion     % Test
-libraryDependencies += "org.scalatestplus.play" %% "scalatestplus-play" % "4.0.1" % Test
-
-// Test Database
```
Comment (marcospereira) [inline build.sbt:35]: Not used.

File: build.sbt [hunk 3]
```diff
@@ -18,26 +18,13 @@ val `play-java-grpc-example` = (project in file("."))
...
-libraryDependencies += "com.h2database" % "h2" % "1.4.197"
-
-// Testing libraries for dealing with CompletionStage...
```
Comment (marcospereira) [inline build.sbt:38]: Not used.

File: test/example/myapp/helloworld/HelloFunctionalTest.java [hunk 4]
```diff
@@ -19,33 +23,28 @@
 import static org.junit.Assert.*;
 import static play.inject.Bindings.*;
 
-public final class HelloFunctionalTest {
-  private final TestServerFactory testServerFactory = new DefaultTestServerFactory();
+public final class HelloFunctionalTest extends WithApplication {
```
Comment (marcospereira) [inline test/example/myapp/helloworld/HelloFunctionalTest.java:26]: Can use `WithApplication` and let Play start/stop the `app` for you.

**Review by ignasi35 — APPROVED** [review]
❤️

**Pull request merged**